python/src/scipp/plot/controller3d.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @author Neil Vaytet
+
+# Scipp imports
+from .controller import PlotController
+from .._utils import name_with_unit
+from .._scipp import core as sc
+
+# Other imports
+import numpy as np
+
+
+class PlotController3d(PlotController):
+    def __init__(self,
+                 scipp_obj_dict=None,
+                 pixel_size=None,
+                 positions=None,
+                 **kwargs):
+
+        super().__init__(scipp_obj_dict=scipp_obj_dict,
+                         positions=positions,
+                         **kwargs)
+
+        self.positions = positions
+        self.pos_axparams = {}
+
+        # If positions are specified, then the x, y, z points positions can
+        # never change
+        if self.positions is not None:
+            coord = scipp_obj_dict[self.name].coords[self.positions]
+            for xyz in "xyz":
+                x = getattr(sc.geometry, xyz)(coord)
+                self.pos_axparams[xyz] = {
+                    "lims": [
+                        sc.min(x).value - 0.5 * pixel_size,
+                        sc.max(x).value + 0.5 * pixel_size
+                    ],
+                    "label":
+                    name_with_unit(coord, name=xyz.upper())
+                }
+
+    def _get_axes_parameters(self):
+        axparams = {}
+        if self.positions is not None:
+            axparams = self.pos_axparams
+        else:
+            axparams = super()._get_axes_parameters()
+
+        axparams["centre"] = [
+            0.5 * np.sum(axparams['x']["lims"]),
+            0.5 * np.sum(axparams['y']["lims"]),
+            0.5 * np.sum(axparams['z']["lims"])
+        ]
+
+        axparams["box_size"] = np.array([
+            axparams['x']["lims"][1] - axparams['x']["lims"][0],
+            axparams['y']["lims"][1] - axparams['y']["lims"][0],
+            axparams['z']["lims"][1] - axparams['z']["lims"][0]
+        ])
+        return axparams
+
+    def get_positions_array(self):
+        return self.model.get_positions_array()
```

Comment:
Not having checked how this is used, this public method feels odd: Other methods on the controller instruct it to do/update something, whereas this provides access to the model directly?

Reply:
This has been removed